### DISCUSSION TITLE
Avoid giving admin privileges after restart on MacOS

### DISCUSSION THREAD

**Discussion by AlbertoFemenias** [rancher-sandbox/rancher-desktop#1814] [Q&A] [answered]
Hi, after reading many issues and discussions in the repo, I start this new discussion seeking some help. 

I work with a personal MacBook Air (M1, 2020, MacOS BigSur 11.6.4) and use rancher desktop 1.1.1 as my container manager. We would like to extend its use to all corporate computers but we have a big caveat, after each restart this prompt asks for root permissions:

<img width="562" alt="Screenshot 2022-03-14 at 18 42 39" src="https://user-images.githubusercontent.com/56123138/158333763-907fd7e2-969e-4087-bcc6-8d9d2af3b4cd.png">

The main problem is that workers with corporate Macs do not have admin credentials. 
Is there a workaround to avoid giving sudo permissions on each restart?

Thanks in advance!

**Comment by jandubois** [comment, marked as answer]
The reason for the prompt is that macOS completely deletes the contents of `/var/run` on each reboot.

We are getting rid of `/var/run/rancher-desktoplima` in the next release for that reason. We are going to create the files directly inside the `/var/run` directory (with a `rancher-desktop` prefix in the filename) to avoid having to recreate the directory.

We have not yet implemented a way to create the docker socket in a user-owned location, so this will continue to prompt for sudo access unless you switch to containerd and nerdctl.

See also #1820 for some work-in-progress. It mentions the need for a followup PR to move the docker socket location. Unfortunately it looks like these changes will not yet make it into the next release (which has become urgent due to the recent breakage with macOS Monterey 12.3, e.g. #1815).

**Reply by AlbertoFemenias** [reply]
Thanks @jandubois, you are doing a great job making rancher desktop stable in such short time!! 
Looking forward to the next releases 😄

**Reply by schlueter** [reply]
This is still an issue with _Rancher Desktop 1.5.1_. Is there a timeline or proposed version for when this should no longer be an issue?

**Reply by jandubois** [reply]
> Is there a timeline or proposed version

Unfortunately there isn't yet. Realistically I would guess some time in the first half of next year, given the people we have available and other things in progress. I hope it will be early next year, but I can't promise it.

**Reply by mahiki** [reply]
I'm just glad its being worked on, I'll put up with it for now thanks folks you are the best.

**Reply by sourcecodemage** [reply]
Is this getting close to completion? I'm on v1.8.1 and my work is still getting interrupted to grant admin access to Rancher Desktop. MacOS 13.2.1

**Reply by jandubois** [reply]
Unfortunately it isn't close yet, due to some contention for the time of the developers who would work on that.

Note that you can disable admin access as long as you don't need a routable IP address for your VM. You still get port forwarding to the host. And you have to make sure you run `docker context use rancher-desktop` to make sure docker finds the socket in the non-default location (assuming you use Moby).

**Reply by huyz** [reply]
@jandubois Ah good to know!

**Reply by jsfr** [reply]
Just to confirm, a version resolving this hasn't been released yet right?

**Reply by jandubois** [reply]
> Just to confirm, a version resolving this hasn't been released yet right?

Depends on what you mean by "this", but yes, the status is still the same as described in https://github.com/rancher-sandbox/rancher-desktop/discussions/1814#discussioncomment-5493330.

If you need an externally routable IP address, or the docker socket in the default location, then you still need admin access, which means you will get prompted for your password once after each reboot.

Note that the vast majority of users don't have these requirements, so admin mode is no longer the default in Rancher Desktop.

**Reply by sebastianhaberey** [reply]
I am also suffering from this, it is really annoying to have to do this every day. I was thinking I need this for IntelliJ to connect to Rancher Desktop, as was suggested in [this issue](https://youtrack.jetbrains.com/issue/IDEA-287703/docker-integration-support-for-rancher-desktop#focus=Comments-27-6687367.0-0).

But @jandubois am I reading this right: without administrative access, Rancher Desktop would simply place the socket somewhere else? Where would that be - I'd like to try to reconfigure IntelliJ to that location and run RD without administrative access.

**Reply by jandubois** [reply]
**Update** This is kind of obsolete; please see me next comment below on how to configure the plugin.


You can check the location of the socket using `docker context ls`, e.g.

```console
$ docker context ls
NAME                DESCRIPTION                               DOCKER ENDPOINT                                   ERROR
default             Current DOCKER_HOST based configuration   unix:///var/run/docker.sock
rancher-desktop *   Rancher Desktop moby context              unix:///Users/jan/.rd/docker.sock
```

Ideally IntelliJ should use the Docker context to determine the location of the socket, but I have no idea if it does.

An older mechanism to specify the location of the socket to client software is the `DOCKER_HOST` environment variable. This will also override the current context. See

```console
$ DOCKER_HOST=unix:///hello/world docker context ls
NAME              DESCRIPTION                               DOCKER ENDPOINT                                   ERROR
default *         Current DOCKER_HOST based configuration   unix:///hello/world
rancher-desktop   Rancher Desktop moby context              unix:///Users/jan/.rd/docker.sock
Warning: DOCKER_HOST environment variable overrides the active context. To use a context, either set the global --context flag, or unset DOCKER_HOST environment variable.
```

Now `rancher-desktop` is not the active context anymore, and the endpoint for the `default` context has changed.

I don't know if IntelliJ would support the env variable either, but for desktop apps on macOS that would be hard to configure because the app is launched via `launchd` and does not inherit the user's shell environment settings.

**Reply by jandubois** [reply]
I assume the IntelliJ plugin is the same as the one for GoLand. I just tested it, and the "Docker for Mac" setting has a dropdown that lets you select a different context. Pick `rancher-desktop` here, and it should just work:

![CleanShot 2024-02-14 at 00 25 13@2x](https://github.com/rancher-sandbox/rancher-desktop/assets/78947/a17963f9-c8ab-4592-9522-a517d9f95482)

**Comment by sebastianhaberey** [comment]
@jandubois thanks for the thorough explanation! I was able to find the socket using `docker context ls` but like you said it wasn't even necessary because there was already an option for Rancher Desktop in IntelliJ:

<img width="613" alt="Bildschirmfoto 2024-02-15 um 07 58 22" src="https://github.com/rancher-sandbox/rancher-desktop/assets/4152849/39840023-6717-49cd-912d-5ebc6f1a88c2">

Consequently I was able to disable administrative access permanently in the RD startup screen and use RD without the daily popup.
